client/layout/guided-tours/index.js

```diff
@@ -15,14 +15,20 @@ import QueryPreferences from 'components/data/query-preferences';
 import RootChild from 'components/root-child';
 import { getGuidedTourState } from 'state/ui/guided-tours/selectors';
 import { getLastAction } from 'state/ui/action-log/selectors';
-import { getSectionName, isSectionLoading } from 'state/ui/selectors';
-import { nextGuidedTourStep, quitGuidedTour } from 'state/ui/guided-tours/actions';
+import { getInitialQueryArguments, getSectionName, isSectionLoading } from 'state/ui/selectors';
+import { nextGuidedTourStep, quitGuidedTour, resetGuidedTourHistory } from 'state/ui/guided-tours/actions';
 
 class GuidedTours extends Component {
 	shouldComponentUpdate( nextProps ) {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
+	componentWillReceiveProps( nextProps ) {
```

Comment:
Side note on the choice of `componentWillReceiveProps`: we're fetching `tour` from the initial query arguments, so `props.tour` shouldn't ever change in the lifetime of our application. Have you tried performing the check in, say, `componentWillMount`?

Reply:
For both `componentWillMount` and `componentDidMount`, the `queryArgs` in `props` are still false when they happen. Not sure whether there's a way to make that play nice — and ideas?

Reply:
I did find this weird too and `componentWillMount` was even my first suggestion before it was switched to `componentWillReceiveProps` because it really wasn't working that way.

Reply:
Yeah, that's to be expected, unfortunately, due to how we populate `state.ui.queryArguments`:

https://github.com/Automattic/wp-calypso/blob/5a0386b3a3e74275951a69829d911160ca1bb9ef/client/boot/common.js#L121-L130